components/logins/src/sync/engine.rs

```diff
@@ -504,6 +506,22 @@ impl SyncEngine for LoginsSyncEngine {
     }
 }
 
+// This seems misplaced - but everywhere else a mirror row is read we don't need the unknown
+// fields because they are, um, unknown :) So we never merge or inspect them, just send
+// whatever we have in outgoing.
+fn unknown_fields_from_row(
+    row: &rusqlite::Row<'_>,
+) -> Result<Option<super::payload::UnknownFields>> {
+    Ok(match row.get::<_, Option<String>>("unknown_fields")? {
+        None => None,
+        Some(s) => match serde_json::from_str(&s)? {
+            serde_json::Value::Object(m) => Some(m),
+            // ???? Should maybe log/report an error?
```

Comment:
I like reporting an error.  It seems like the data was corrupted in some way and now we're going to drop it.

Reply:
If the data was corrupted, it's more likely the `from_str()` would fail. I'm inclined to think that it would actually be OK to panic if we ended up with valid json that wasn't an object - we have a number of other places we panic in clearly impossible cases and no evidence of actually hitting them.

(On the fence still about the `from_str()` failing though - I mean, we don't really try and ignore other corruption scenarios. autofill stores json in a string in the db and I'm fairly sure it just has a normal error return if that fails)

Reply:
If one record is broken, we can not panic and stop processing all others. I'm sure you meant to return an error for broken record.

Reply:
This is almost a philosophical question :) We are talking about our local DB and sqlite is quite a loosely typed DB. To my mind, this problem would be analogous to, say, finding that a column our schema asserts is a number is actually a string, or a non-nullable column has a null - all things that could happen with corrupt databases and all things that would bring us down today. I'd agree we can't panic if this was about incoming data, but it's just us trusting both our implementation and our database. Some sync engines have explicit `assert()` statements in similar "impossible" cases, and I'm generally fine with that, and indeed they have saved us in the past (ie, the one case I can think of when an assert failed, we agreed that outcome was better than had the assert not existed and we charged on regardless in the face of a real bug)

Reply:
eg, https://www.sqlite.org/faq.html#q3